GateKeeper.cpp

```diff
@@ -383,6 +385,105 @@ namespace pEp {
         return enabled;
     }
 
+    bool GateKeeper::is_update_available() {
+        bool result = true;
+        NTSTATUS status = BCryptOpenAlgorithmProvider(&hAES, BCRYPT_AES_ALGORITHM, MS_PRIMITIVE_PROVIDER, 0);
+        assert(status == 0);
+        result = status == 0;
+        if (status)
+            goto closing;
+        status = BCryptSetProperty(hAES, BCRYPT_CHAINING_MODE, (PUCHAR)BCRYPT_CHAIN_MODE_GCM, sizeof(BCRYPT_CHAIN_MODE_GCM), 0);
+        if (status)
+            goto closing;
+
+        status = BCryptOpenAlgorithmProvider(&hRSA, BCRYPT_RSA_ALGORITHM, MS_PRIMITIVE_PROVIDER, 0);
+        assert(status == 0);
+        result = status == 0;
+        if (status)
+            goto closing;
+
+        internet = InternetOpen(_T("pEp"), INTERNET_OPEN_TYPE_PRECONFIG, NULL, NULL, 0);
+        if (!internet)
+            goto closing;
+
+        {
+            product_list products = registered_products();
+            DWORD context = 0;
+           
+            result = products.size() > 0;
+            for (auto i = products.begin(); i != products.end(); i++) {
+                try {
+                    result = result && is_product_url_exists(*i, context++);

```

Comment:
this code only works if we assume there is only 1 planck product installed in the same machine. Otherwise it will fail, i mean, in case there is an url which does not exist for some product then the result will be always false which is not correct since there may be some update for other products. As talked with @RobertoVerdesoto , since we are assuming that from now one will be only 1 product installed in the system (the Outlook plugin) then this code is correct since in the product list there will always be only one product at most.